### PR TITLE
Introduction to terminal multiplexers using tmux

### PR DESCRIPTION
**Pull request by willfurnass**
- [x] Issues that arise when not using terminal multiplexers
- [x] Causes of some of those issues
- [x] Detaching and attaching to sessions with examples
- [x] Multiplexing using windows with examples
- [x] Multiplexing using panes
- [x] Copy and paste
- [x] Killing locked up windows
- [x] Scroll buffer
- [x] Usage on HPC
    - [x] where to run tmux server
    - [x] resource footprint
    - [x] reusing existing tmux sessions to minimise footprint on login nodes (`tmux new-session -A -s main`)
- [x] List things not covered (tmux fingers, custom config, sharing sessions)
- [x] Check examples aren't affected by / specific to [my tmux config](https://github.com/willfurnass/dotfiles/blob/master/tmux/.tmux.conf)

### CODE REVIEW

File: site/posts/tmux-intro.rst [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+--------------------------
+
+Terminal Multiplexers can be useful if doing `interactive work <http://docs.hpc.shef.ac.uk/en/latest/hpc/scheduler/sge.html#interactive-sessions)>`__ 
+on a `HPC <https://en.wikipedia.org/wiki/Supercomputer>`__ cluster such as the University of Sheffield clusters `ShARC and Iceberg <http://docs.hpc.shef.ac.uk/>`__
+(assuming that you don't need a `GUI <https://en.wikipedia.org/wiki/Graphical_user_interface>`__).
+
+On ShARC and Iceberg can:
+
+#. Start a tmux or GNU Screen session on a login node;
+#. Start an interactive job using `qrshx or qrshx <http://docs.hpc.shef.ac.uk/en/latest/hpc/scheduler/sge.html>`__;
```
Comment (anthonybrookfield) [inline site/posts/tmux-intro.rst]: "qrshx or qrshx"?

File: site/posts/tmux-intro.rst [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+Your interactive job (on a cluster worker node) will be terminated by the cluster's job scheduler after a fixed amount of time (the default is 8 hours)
+but your tmux/Screen session was started on a login node so is outside the control of the cluster and
+will keep running indefinitely unless you kill it.
+
+Each tmux/Screen session requires memory on the login node (which is used by all users) so to be a good HPC citizen you should:
+
+-  **Kill your tmux/Screen session when you no longer need it** (tmux/Screen will exit when you close all windows)
+-  **Only start as many tmux/Screen sessions on the login node as you need (ideally 1)**
+
```
Comment (anthonybrookfield) [inline site/posts/tmux-intro.rst]: add a comment about not leaving interactive sessions hanging around when not being used.... (due to limited availability)?

File: site/posts/tmux-intro.rst [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+However, note that you can have nested tmux sessions 
+(with **<prefix><prefix> <key>** used to send tmux commands to the 'inner' tmux session).
+
+Being a good HPC citizen
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Your interactive job (on a cluster worker node) will be terminated by the cluster's job scheduler after a fixed amount of time (the default is 8 hours)
+but your tmux/Screen session was started on a login node so is outside the control of the cluster and
+will keep running indefinitely unless you kill it.
```
Comment (anthonybrookfield) [inline site/posts/tmux-intro.rst]: Warn about how [hpc name].shef.ac.uk may be repointed to another login node if the current login node needs rebooting/maintenance?  we'll also kill without warning any long running login node processes after draining the login nodes of users....

**Comment by mikecroucher** [comment]
Mind if we merge this? Should be enough for Petar -- the academic who wants this. Can always fix any other minor problems post publication.

**Comment by willfurnass** [comment]
@mikecroucher done; see http://rse.shef.ac.uk/blog/tmux-intro/

**Pull request merged**